src/fits.rs

```diff
@@ -0,0 +1,190 @@
+            let mut new_row = vec![0.0; 63];
+            let start = (63 - naxis1) / 2;
+            for i in 0..naxis1 as i32 {
+                new_row[(start + i) as usize] = row[i as usize];
+            }
+            new_image_data.extend(new_row);
+        }
+        image_data = new_image_data;
+    }
+
+    if naxis2 != 63 {
+        let mut new_image_data = vec![0.0; 63 * 63];
+        let start = (63 - naxis2) / 2;
+        for i in 0..naxis2 as i32 {
+            for j in 0..63 {
+                new_image_data[((start + i) * 63 + j) as usize] =
+                    image_data[(i * naxis1 + j) as usize];
+            }
+        }
+        image_data = new_image_data;
+    }
+
+    Ok(image_data)
+}
+
+/// Normalizes an image by replacing NaNs with 0s and infinities with the maximum finite value
+/// Then, it computes the Frobenius norm of the image
+/// and normalizes the image by dividing each pixel by the norm
+pub fn normalize_image(image: Vec<f32>) -> Result<Vec<f32>, Box<dyn std::error::Error>> {
+    // first, replace all NaNs with 0s
+    // and infinities with the maximum finite value
+    let mut normalized = vec![];
+    for pixel in image {
+        if pixel.is_nan() {
+            //println!("Found a NaN");
+            normalized.push(0.0);
+        } else if pixel.is_infinite() {
+            println!("Found an infinity");
+            // f32MIN if the number is negative, f32MAX if the number is positive
+            normalized.push(if pixel.is_sign_negative() {
+                f32::MIN
+            } else {
+                f32::MAX
+            });
+        } else {
+            normalized.push(pixel);
+        }
+    }
+    // then, compute the norm of the vector, which is the Frobenius norm of this array
+    // so a 2-norm of a vector is the square root of the sum of the squares of the elements (in absolute value)
+    let norm: f32 = normalized.iter().map(|x| x.powi(2)).sum::<f32>().sqrt();
+
+    normalized = normalized.iter().map(|x| x / norm).collect();
+    Ok(normalized)
+}
+
+/// Prepares a cutout image for ML models
+/// It reads the image from the alert document
+/// decompresses it, normalizes it and returns it as a vector of f32
+/// The image is expected to be a 2D array of floats
+fn prepare_cutout(cutout: Vec<u8>) -> Result<Vec<f32>, Box<dyn std::error::Error>> {
+    let cutout = buffer_to_image(cutout).unwrap();
+    let cutout = normalize_image(cutout).unwrap();
+    Ok(cutout)
+}
+
+/// Prepares a triplet of cutouts for ML models
+/// It reads the images from the alert document
+/// decompresses them, normalizes them and returns them as a tuple of vectors of f32
+/// The images are expected to be 2D arrays of floats
+pub fn prepare_triplet(
+    alert_doc: &mongodb::bson::Document,
+) -> Result<(Vec<f32>, Vec<f32>, Vec<f32>), Box<dyn std::error::Error>> {
+    let cutout_science = alert_doc
+        .get_binary_generic("cutoutScience")
+        .unwrap()
```

Comment:
I'm fine leaving these `unwraps` for now. It would be great to have a ticket reminding us to come back to add error handling, though.